anvio/synteny.py

```diff
@@ -323,8 +324,25 @@ def annotate_window(self, window):
         # Annotate window based on user input
         gene_annotation_dict = {}
         for annotation_source, annotations_dict in self.annotation_source_dict.items():
-            ngram_annotation = tuple([annotations_dict[g] for g in window])
-            gene_annotation_dict[annotation_source] = ngram_annotation 
+            if self.take_first_cog:
+                if annotation_source == 'COG_FUNCTION':
```

Comment:
you don't need to test if the annotation_source is COG here. if you remove this `if` statement (and fix the indentation, of course), the rest of the code should work even if the user chose `KOFams` as `--ngram-source`.

Reply:
Good idea, I was thinking too specific.